src/tree/hist/histogram.h

```diff
@@ -190,6 +193,17 @@ class HistogramBuilder {
           reinterpret_cast<double *>(this->hist_[first_nidx].data()), n);
     }
 
+    if (is_distributed_ && is_col_split_ && is_secure_) {
+      // Under secure vertical mode, we perform allgather for all nodes
+      CHECK(!nodes_to_build.empty());
+      // in theory the operation is AllGather, but with current system functionality,
```

Comment:
what's the issue with the current system? I'm working on the network module, maybe I can help?

Reply:
sorry for the confusion, what I mean is that AllReduce is the most direct and effective solution to handle this computation (all histograms have the same length, those "empty" slots hold value 0, so AllReduce can update them in place).

Reply:
comment updated